chameleonultragui/lib/gui/components/slotsettings.dart

```diff
@@ -0,0 +1,349 @@
+import 'package:chameleonultragui/bridge/chameleon.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:chameleonultragui/main.dart';
+
+class SlotSettings extends StatefulWidget {
+  final int slot;
+  final dynamic refresh;
+
+  const SlotSettings({super.key, required this.slot, required this.refresh});
+
+  @override
+  SlotSettingsState createState() => SlotSettingsState();
+}
+
+class SlotSettingsState extends State<SlotSettings> {
+  List<bool> selectedGen1aMode = <bool>[true, false];
+  List<bool> selectedGen2Mode = <bool>[true, false];
+  List<bool> selectedAntiColl = <bool>[true, false];
+  List<bool> selectedDetection = <bool>[true, false];
+  List<bool> selectedWriteMode = <bool>[false, false, false, false];
+  List<bool> selectedEnabled = <bool>[true, false];
+  bool firstRun = true;
+  String hfName = "";
+  String lfName = "";
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  Future<void> fetchInfo() async {
+    var appState = context.read<MyAppState>();
+    var connection = ChameleonCom(port: appState.connector);
+
+    if (hfName.isEmpty) {
+      try {
+        hfName = (await connection.getSlotTagName(
+                widget.slot, ChameleonTagFrequiency.hf))
+            .trim();
+      } catch (_) {}
+
+      if (hfName.isEmpty) {
+        hfName = "Empty";
+      }
+
+      setState(() {});
+    }
+
+    if (lfName.isEmpty) {
+      try {
+        lfName = (await connection.getSlotTagName(
+                widget.slot, ChameleonTagFrequiency.lf))
+            .trim();
+      } catch (_) {}
+
+      if (lfName.isEmpty) {
+        lfName = "Empty";
+      }
+
+      setState(() {});
+    }
+
+    if (firstRun) {
```

Comment:
initialize in init state or somewhere else, "first run" is ugly and not how its supposed to work...